massa-protocol-worker-2/src/messages.rs

```diff
@@ -1,11 +1,231 @@
 use crossbeam::channel::Sender;
-use massa_serialization::{DeserializeError, Deserializer, U64VarIntDeserializer};
+use massa_serialization::{DeserializeError, Serializer, Deserializer, U64VarIntDeserializer, U64VarIntSerializer};
+use num_enum::{TryFromPrimitive, IntoPrimitive};
 use peernet::{
     error::{PeerNetError, PeerNetResult},
-    messages::MessagesHandler as PeerNetMessagesHandler,
+    messages::{MessagesHandler as PeerNetMessagesHandler, MessagesSerializer as PeerNetMessagesSerializer},
     peer_id::PeerId,
 };
 
+use crate::handlers::{peer_handler::{PeerManagementMessage, PeerManagementMessageSerializer}, block_handler::{BlockMessage, BlockMessageSerializer}, endorsement_handler::{EndorsementMessage, EndorsementMessageSerializer}, operation_handler::{OperationMessage, OperationMessageSerializer}};
+
+pub enum Message {
+    BlockMessage(BlockMessage),
+    EndorsementMessage(EndorsementMessage),
+    OperationMessage(OperationMessage),
+    PeerManagementMessage(PeerManagementMessage),
+}
+
+impl From<BlockMessage> for Message {
```

Comment:
Maybe it would be nice to create some kind of macro to generate all these boring code ?

Reply:
Added a comment